skrifa/src/bitmap.rs

```diff
@@ -0,0 +1,496 @@
+            // TODO: Using Size::unscaled and LocationRef::default should be fine here, right? We need
+            // it for calculating the bounding box and left-side-bearing, which we then provide in
+            // font units. The location_ref could in theory be problematic, but I don't think there
+            // are variable fonts that use `sbix`? Not sure what the best solution here is, should
+            // we let the user provide a location?
+            BitmapFormat::Sbix => StrikesKind::Sbix(
+                font.sbix().ok()?,
+                font.glyph_metrics(Size::unscaled(), LocationRef::default()),
+            ),
+            BitmapFormat::Cbdt => {
+                StrikesKind::Cbdt(CbdtTables::new(font.cblc().ok()?, font.cbdt().ok()?))
+            }
+            BitmapFormat::Ebdt => {
+                StrikesKind::Ebdt(EbdtTables::new(font.eblc().ok()?, font.ebdt().ok()?))
+            }
+        };
+        Some(Self(kind))
+    }
+
+    /// Returns the format representing the underlying table for this set of
+    /// strikes.
+    pub fn format(&self) -> Option<BitmapFormat> {
+        match &self.0 {
+            StrikesKind::None => None,
+            StrikesKind::Sbix(..) => Some(BitmapFormat::Sbix),
+            StrikesKind::Cbdt(..) => Some(BitmapFormat::Cbdt),
+            StrikesKind::Ebdt(..) => Some(BitmapFormat::Ebdt),
+        }
+    }
+
+    /// Returns the number of available strikes.
+    pub fn len(&self) -> usize {
+        match &self.0 {
+            StrikesKind::None => 0,
+            StrikesKind::Sbix(sbix, _) => sbix.strikes().len(),
+            StrikesKind::Cbdt(cbdt) => cbdt.location.bitmap_sizes().len(),
+            StrikesKind::Ebdt(ebdt) => ebdt.location.bitmap_sizes().len(),
+        }
+    }
+
+    /// Returns true if there are no available strikes.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Returns the strike at the given index.
+    pub fn get(&self, index: usize) -> Option<BitmapStrike<'a>> {
+        let kind = match &self.0 {
+            StrikesKind::None => return None,
+            StrikesKind::Sbix(sbix, metrics) => {
+                StrikeKind::Sbix(sbix.strikes().get(index).ok()?, metrics.clone())
+            }
+            StrikesKind::Cbdt(tables) => StrikeKind::Cbdt(
+                tables.location.bitmap_sizes().get(index).copied()?,
+                tables.clone(),
+            ),
+            StrikesKind::Ebdt(tables) => StrikeKind::Ebdt(
+                tables.location.bitmap_sizes().get(index).copied()?,
+                tables.clone(),
+            ),
+        };
+        Some(BitmapStrike(kind))
+    }
+
+    /// Returns the best matching glyph for the given size and glyph
+    /// identifier.
+    ///
+    /// In this case, "best" means a glyph of the exact size, nearest larger
+    /// size, or nearest smaller size, in that order.
+    pub fn glyph_for_size(&self, size: Size, glyph_id: GlyphId) -> Option<BitmapGlyph<'a>> {
```

Comment:
@drott this behavior is slightly different from what currently exists in Skia in that it only considers strikes that actually contain the requested glyph. Since coverage can technically be sparse, I believe this is a more robust solution. We can add a method for selecting strikes without considering glyph coverage if you’d prefer to retain current behavior.

@bungeman I believe you raised a concern about this at some point?

Reply:
Works for me as long as we keep the "first larger, then smaller" logic.